File: play/load.py
from __future__ import unicode_literals

import tensorflow as tf


MODEL_PREFIX = "chessbot"


def load_graph():
    with tf.gfile.GFile("chessbot.pb", "rb") as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read())

    with tf.Graph().as_default() as graph:
        tf.import_graph_def(
            graph_def,
            input_map=None,
            return_elements=None,
            name=MODEL_PREFIX,
            op_dict=None,
            producer_op_list=None
        )
        return graph


def load_session(graph):
    session = tf.Session(graph=graph)
    session.x = graph.get_tensor_by_name('chessbot/prediction_inputs:0')
    session.y = dict(class_ids=graph.get_tensor_by_name('chessbot/dnn/head/predictions/class_ids:0'))
    return session

File: play/load.go
package play

import (
	"io/ioutil"

	tf "github.com/tensorflow/tensorflow/tensorflow/go"
)

type Model struct {
	graph   *tf.Graph
	session *tf.Session
	x       tf.Output
	y       tf.Output
}

func (m *Model) Evaluate(input BoardInput) (int64, error) {
	inputTensor, err := tf.NewTensor(input)
	if err != nil {
		return 255, err
	}
	tensors, err := m.session.Run(
		map[tf.Output]*tf.Tensor{
		    m.x: inputTensor,
		},
		[]tf.Output{m.y},
		nil,
	)
	if err != nil {
		return 255, err
	}
	output := (tensors[0].Value().([]int64))[0]
	return output, nil
}

func (m *Model) Close() error {
	return m.session.Close()
}

func LoadModel(modelPath string) (*Model, error) {
	modelBytes, err := ioutil.ReadFile(modelPath)
	if err != nil {
		return nil, err
	}

	graph := tf.NewGraph()
    	if err = graph.Import(modelBytes, "chessbot"); err != nil {
        	return nil, err
    	}

	session, err := tf.NewSession(graph, nil)
    	if err != nil {
        	return nil, err
    	}

	model := Model{
		graph: graph,
		session: session,
		x: graph.Operation("chessbot/prediction_inputs").Output(0),
		y: graph.Operation("chessbot/dnn/head/predictions/class_ids").Output(0),
	}
    	return &model, nil
}
